src/components/classrooms/StudentsTable.js

```diff
@@ -0,0 +1,352 @@
+      rows,
+    } = this.state;
+
+    const { classes, classroom } = this.props;
+
+    const isSelected = name => selected.indexOf(name) !== -1;
+
+    const headCells = this.buildHeadCells();
+
+    const emptyRows =
+      rowsPerPage - Math.min(rowsPerPage, rows.length - page * rowsPerPage);
+
+    return (
+      <div className={classes.root}>
+        <Paper className={classes.paper}>
+          <TableToolbar
+            numSelected={selected.length}
+            classroomName={classroom.get('name')}
+          />
+          <TableContainer>
+            <Table size={dense ? 'small' : 'medium'}>
+              <TableHeader
+                headCells={headCells}
+                classes={classes}
+                numSelected={selected.length}
+                order={order}
+                orderBy={orderBy}
+                onSelectAllClick={this.handleSelectAllClick}
+                onRequestSort={this.handleRequestSort}
+                rowCount={rows.length}
+              />
+              <TableBody id={CLASSROOM_TABLE_BODY_ID}>
+                {stableSort(rows, getComparator(order, orderBy))
+                  .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+                  .map(row => {
+                    const isItemSelected = isSelected(
+                      row[USERNAME_HEAD_CELL_ID]
+                    );
+
+                    const handleOnClick = event => {
+                      this.handleClick(event, row[USERNAME_HEAD_CELL_ID]);
+                    };
+
+                    return (
+                      <TableRow
+                        data-name={row[USERNAME_HEAD_CELL_ID]}
+                        hover
+                        onClick={handleOnClick}
+                        role="checkbox"
+                        aria-checked={isItemSelected}
+                        tabIndex={-1}
+                        key={row[USERNAME_HEAD_CELL_ID]}
+                        selected={isItemSelected}
+                      >
+                        <TableCell padding="checkbox">
+                          <Checkbox checked={isItemSelected} color="primary" />
+                        </TableCell>
+                        <TableCell>{row[USERNAME_HEAD_CELL_ID]}</TableCell>
+                        {headCells
+                          // remove first element and last element: username and operations
+                          .slice(1, headCells.length - 1)
+                          // display all other data depending on headcells
+                          .map(({ id }) => (
+                            <TableCell key={id} align="right">
+                              {row[id]}
+                            </TableCell>
+                          ))}
+                        {this.renderUserOperations(row)}
+                      </TableRow>
+                    );
+                  })}
+                {emptyRows > 0 && (
+                  <TableRow style={{ height: (dense ? 33 : 53) * emptyRows }}>
```

Comment:
Yeah, I'm thinking that the dense switch/condition just clutters things a bit. We should choose one or the other.